components/director/internal/domain/application/service.go

```diff
@@ -93,13 +97,39 @@ func (s *service) List(ctx context.Context, filter []*labelfilter.LabelFilter, p
 	return s.appRepo.List(ctx, appTenant, filter, pageSize, cursor)
 }
 
-func (s *service) ListByRuntimeID(ctx context.Context, runtimeID string, pageSize *int, cursor *string) (*model.ApplicationPage, error) {
+func (s *service) ListByRuntimeID(ctx context.Context, runtimeID uuid.UUID, pageSize *int, cursor *string) (*model.ApplicationPage, error) {
 	tenantID, err := tenant.LoadFromContext(ctx)
+
 	if err != nil {
 		return nil, errors.Wrapf(err, "while loading tenant from context")
 	}
 
-	return s.appRepo.ListByRuntimeID(ctx, tenantID, runtimeID, pageSize, cursor)
+	tenantUUID, err := uuid.Parse(tenantID)
+	if err != nil {
+		return nil, errors.New("tenantID is not UUID")
+	}
+
+	label, err := s.labelRepo.GetByKey(ctx, tenantID, model.RuntimeLabelableObject, runtimeID.String(), model.ScenariosKey)
```

Comment:
you changed runtimeID from string to UUID, but in repo you have string and again you write: `runtimeID.String()`